src/material/tabs/tab-nav-bar/tab-nav-bar.ts

```diff
@@ -151,6 +183,7 @@ export abstract class _MatTabNavBase
   templateUrl: 'tab-nav-bar.html',
   styleUrls: ['tab-nav-bar.css'],
   host: {
+    '[attr.role]': 'tabPanel ? "tablist" : null',
```

Comment:
@crisbeto This - as well as some of the `aria-*` attribute host bindings changes below - are breaking changes. For example, if a consumer has added `role="<something>"` to this element, then after this PR is submitted, this host binding will override their specified `role` and set it to `null` (as no `tabPanel` will have been provided). We can avoid this breaking change by modifying this logic to check to see if a `role` has been provided, and if so, use that. For example:

```
'[attr.role]': 'existingRole || (tabPanel ? "tablist" : null)'
```

We'd want to do the same to the other added `aria-*` host bindings. WDYT?

Reply:
Okay.

Reply:
Updated the PR per our conversation to fallback to the existing ARIA host attributes if no `tabPanel` is provided, as opposed to removing those attributes with a `null` attribute binding. This makes this PR now truly 100% opt-in & backwards compatible. Consumer code should not change if `tabPanel` isn't provided. This makes it _much_ easier to land internally.

After this PR is merged in, I'll:
1. Migrate the internal instances of consumers overriding ARIA host attributes to the new `tabPanel` input & associated component.
2. Remove the code in this PR that "falls back to the existing ARIA host attributes if no `tabPanel` is provided" (as it calls the DOM directly and not something we want in the component long-term).
3. Migrate the remaining consumer instances onto the new `tabPanel` input & associated component.
4. Remove the code in this PR that falls back to the ARIA link / navigation landmark pattern if no `tabPanel` is provded.